aptos-move/framework/src/natives/object.rs

```diff
@@ -39,7 +42,9 @@ fn native_exists_at(
     let type_ = ty_args.pop().unwrap();
     let address = safely_pop_arg!(args, AccountAddress);
 
-    context.charge(gas_params.base_cost)?;
+    // We don't need to charge for address as it's fixed length (so can be included in base).
+    context
+        .charge(gas_params.base + gas_params.per_type * NumArgs::new(u64::from(type_.size())))?;
```

Comment:
Two problems here
1. You are using the legacy `.size()`, which has been completely eliminated on our side
2. The unit is wrong

I think we may need to define a new AbstractTypeSize, similar to what I did with AbstractValueSize

Reply:
Is the `calc_abstract_size` parameter to the event natives an example of this in `aptos-move/framework/src/natives/event.rs`?

Reply:
Exactly. Again, please do not use `.size()` here. It's a legacy function that's hardcoded.